lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1625,22 +1633,39 @@ def get_course_discussion_user_stats(
         order_by = order_by or UserOrdering.BY_ACTIVITY
         if order_by != UserOrdering.BY_ACTIVITY:
             raise ValidationError({"order_by": "Invalid value"})
-    course_stats_response = get_course_user_stats(course_key, {
+    params = {
         'sort_key': str(order_by),
         'page': page,
         'per_page': page_size,
-    })
+    }
+    comma_separated_usernames = matched_users_count = matched_users_pages = None
+    if username_search_string:
+        comma_separated_usernames, matched_users_count, matched_users_pages = get_usernames_from_search_string(
+            course_key, username_search_string, page, page_size
+        )
+        if not comma_separated_usernames:
+            return DiscussionAPIPagination(request, 0, 1).get_paginated_response({
+                "results": []
+            })
+        params['usernames'] = comma_separated_usernames
+
+    course_stats_response = get_course_user_stats(course_key, params)
     serializer = UserStatsSerializer(
```

Comment:
I think it would be more clean if we initialise the serialiser after we have added stats. Although this way also works but seems weird that we are passing to serialiser first and then adding empty stats.